packages/grid/_modules_/grid/hooks/root/useGridApiEventHandler.ts

```diff
@@ -10,22 +29,28 @@ export function useGridApiEventHandler(
   options?: { isFirst?: boolean },
 ) {
   const logger = useLogger('useGridApiEventHandler');
+  const { signature } = useGridSelector(apiRef, optionsSelector);
```

Comment:
We could get it from the `GridRootPropsContext` context, but this context is only inserted after the hook.

Reply:
Updated.

Reply:
Am I missing something or is the context provider instantiated in the `GridComponent` which is also the component which calls most of the `useGridApiEventHandler` ?

If so, it will always be undefined even after initialization since its not in a child of the provider.

Reply:
Am, not sure but I changed it and tested it and it seems to work.

Reply:
It should be undefined in every call from a hook that is inside `GridComponent` which is problematic.

@m4theushw even with #2176 it will be the same.
We could quite easily move the the props preprocessing + context provider up if we want to use it in the feature.
It would make sense for me.

Reply:
Well it does work as is but sure.

Reply:
Updated

Reply:
Where does the value come from then ?

Reply:
We don't need this line:

```suggestion
```

The details actually is set in https://github.com/mui-org/material-ui-x/pull/2236/files#diff-940e97227abe24d94ec31abb3fe7f6af369e3fcc9711995fefeca5876e737ee1R28 and passed to the handler.

Reply:
I have unresolved the suggestion, the changes in this file are not needed. It's strange, why do we try to override the behavior of `subscribeEvent`? what happened for the feedback to be missed?